pallets/gear/src/runtime_api.rs

```diff
@@ -310,28 +321,37 @@ where
 
         let payload_arg = payload;
         let mut payload = argument.unwrap_or_default();
-        payload.append(&mut Self::read_state_impl(program_id, payload_arg)?);
+        payload.append(&mut Self::read_state_impl(
+            program_id,
+            payload_arg,
+            allowance_multiplier,
+        )?);
 
         let block_info = BlockInfo {
             height: Self::block_number().unique_saturated_into(),
             timestamp: <pallet_timestamp::Pallet<T>>::get().unique_saturated_into(),
         };
 
+        let gas_allowance = Self::update_gas_allowance(
```

Comment:
same here
don't mess up or down-lying api with explicit allowance num and passed high level multiplier